File: projects/Янчук/3/LABB3NEW3/app/src/main/java/com/example/labb3new3/ui/DetailFragment.java
package com.example.labb3new3.ui;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.labb3new3.R;

public class DetailFragment extends Fragment {

    private static final String ARG_TITLE = "title";
    private static final String ARG_DESC = "desc";

    public static DetailFragment newInstance(String title, String desc) {
        DetailFragment f = new DetailFragment();
        Bundle args = new Bundle();
        args.putString(ARG_TITLE, title);
        args.putString(ARG_DESC, desc);
        f.setArguments(args);
        return f;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_detail, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        TextView tvTitle = view.findViewById(R.id.tvDetailTitle);
        TextView tvDesc = view.findViewById(R.id.tvDetailBody);

        if (getArguments() != null) {
            tvTitle.setText(getArguments().getString(ARG_TITLE));
            tvDesc.setText(getArguments().getString(ARG_DESC));
        }
    }
}

File: projects/Гончеренок/lab3/src_/JsonListApp/app/src/main/java/com/example/jsonlistapp/ui/DetailFragment.kt
package com.example.jsonlistapp.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.jsonlistapp.databinding.FragmentDetailBinding

class DetailFragment : Fragment() {

    private var _binding: FragmentDetailBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentDetailBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val id = arguments?.getInt(ARG_ID) ?: 0
        val user = arguments?.getInt(ARG_USER) ?: 0
        val title = arguments?.getString(ARG_TITLE).orEmpty()
        val body = arguments?.getString(ARG_BODY).orEmpty()

        binding.tvDetailId.text = "ID: $id"
        binding.tvDetailUser.text = "Пользователь: $user"
        binding.tvDetailTitle.text = title
        binding.tvDetailBody.text = body

        binding.btnBack.setOnClickListener {
            parentFragmentManager.popBackStack()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        private const val ARG_ID = "id"
        private const val ARG_USER = "user"
        private const val ARG_TITLE = "title"
        private const val ARG_BODY = "body"

        fun newInstance(id: Int, user: Int, title: String, body: String): DetailFragment {
            val fragment = DetailFragment()
            fragment.arguments = Bundle().apply {
                putInt(ARG_ID, id)
                putInt(ARG_USER, user)
                putString(ARG_TITLE, title)
                putString(ARG_BODY, body)
            }
            return fragment
        }
    }
}